monai/transforms/spatial/array.py

```diff
@@ -1700,3 +1701,50 @@ def __call__(
             grid[:3] += gaussian(offset)[0] * self.magnitude
             grid = self.rand_affine_grid(grid=grid)
         return self.resampler(img, grid, mode=mode or self.mode, padding_mode=padding_mode or self.padding_mode)
+
+
+class CoordConv(Transform):
+    """
+    Appends additional channels encoding coordinates of the input.
+
+    Liu, R. et al. An Intriguing Failing of Convolutional Neural Networks and the CoordConv Solution, NeurIPS 2018.
+    """
+
+    def __init__(
+        self,
+        spatial_channels: Sequence[int],
+    ) -> None:
+        """
+        Args:
+            spatial_channels: the spatial dimensions that are to have their coordinates encoded in a channel and
+                appended to the input. E.g., `(1,2,3)` will append three channels to the input, encoding the
+                coordinates of the input's three spatial dimensions (0 is reserved for the channel dimension).
+        """
+        self.spatial_channels = spatial_channels
+
+    def __call__(self, img: Union[np.ndarray, torch.Tensor]) -> Union[np.ndarray, torch.Tensor]:
+        """
+        Args:
+            img: data to be transformed, assuming `img` is channel first.
+        """
+        if max(self.spatial_channels) > img.ndim - 1:
+            raise ValueError(
+                f"input has {img.ndim-1} spatial dimensions, cannot add CoordConv channel for dim {max(self.spatial_channels)}."
+            )
+        if 0 in self.spatial_channels:
+            raise ValueError("cannot add CoordConv channel for dimension 0, as 0 is channel dim.")
+
+        spatial_dims = img.shape[1:]
+        # pre-allocate memory
+        coord_channels = np.ones((len(self.spatial_channels), *spatial_dims)).astype(img.dtype)
+
+        for i, dim in enumerate(self.spatial_channels):
+            ones = np.ones((1, *spatial_dims))
```

Comment:
Could the calculation of the coordinates be done as:

```python
shape = img.shape[1:]
num_channels = img.shape[0]
coords = sum(np.ogrid[tuple(slice(0, i) for i in shape)])
coords = np.repeat(coords[None], repeat=num_channels, axis=0)
```

Reply:
You could remove the `for s in spatial_channels` by requiring use input to be 0-based on the spatial dimensions (e.g., `(0,1,2)` for `(x,y,z)`).